test/performance/performance-tests-mako.sh

```diff
@@ -37,94 +37,132 @@ initialize --skip-istio-addon --min-nodes=10 --max-nodes=10 --perf --cluster-ver
-kubectl delete configmap config-mako --ignore-not-found=true
+kubectl delete configmap config-mako -n default --ignore-not-found=true
+
+kubectl create configmap config-mako -n default --from-file=test/performance/benchmarks/dataplane-probe/dev.config
 
-kubectl create configmap config-mako --from-file=test/performance/benchmarks/dataplane-probe/dev.config
+run_ytt \
+      -f "${REPO_ROOT_DIR}/test/performance/benchmarks/dataplane-probe/continuous/dataplane-probe-direct.yaml" \
+      -f "${REPO_ROOT_DIR}/test/config/ytt/performance/influx" \
+      -f "${REPO_ROOT_DIR}/test/config/ytt/performance/pods" \
+      --data-value dockerrepo="${KO_DOCKER_REPO}" \
+      --data-value prowtag="${PROW_TAG}" \
+      --data-value influxurl="${INFLUX_URL}" \
+      --data-value influxtoken="${INFLUX_TOKEN}" \
+      --output-files "${ARTIFACTS}/mako-overlay"
 
 ko apply -f test/performance/benchmarks/dataplane-probe/continuous/dataplane-probe-setup.yaml
-ko apply -f test/performance/benchmarks/dataplane-probe/continuous/dataplane-probe-direct.yaml
+ko apply -f "${ARTIFACTS}/mako-overlay/dataplane-probe-direct.yaml"
 
-echo "waiting for test to complete"
-for i in {1..600}; do
-        echo -n "#"
-        sleep 1
-done
-echo ""
+wait_for_test
 
-PODNAME_DPD=`kubectl get pods --selector=job-name=dataplane-probe-deployment --template '{{range .items}}{{.metadata.name}}{{end}}'`
-header "Results from ${PODNAME_DPD}"
-# read results usage: <mako_stub_pod_name> <mako_stub_namespace> <mako_stub_port> <timeout> <retries> <retries_interval> <out_file>
-test/performance/read_results.sh ${PODNAME_DPD} "default" 10001 120 100 10 "${ARTIFACTS}/mako/testrun-dataplane-probe-deployment.csv"
+############################################################################################
+header "Deployment probe performance test"
+kubectl delete job  deployment-probe --ignore-not-found=true
+kubectl delete configmap config-mako -n default --ignore-not-found=true
+kubectl create configmap config-mako -n default --from-file=test/performance/benchmarks/deployment-probe/dev.config
 
-PODNAME_DPI=`kubectl get pods --selector=job-name=dataplane-probe-istio --template '{{range .items}}{{.metadata.name}}{{end}}'`
-header "Results from ${PODNAME_DPI}"
-test/performance/read_results.sh ${PODNAME_DPI} "default" 10001 120 100 10 "${ARTIFACTS}/mako/testrun-dataplane-probe-istio.csv"
+run_ytt \
+      -f "${REPO_ROOT_DIR}/test/performance/benchmarks/deployment-probe/continuous/benchmark-direct.yaml" \
+      -f "${REPO_ROOT_DIR}/test/config/ytt/performance/influx" \
+      -f "${REPO_ROOT_DIR}/test/config/ytt/performance/pods" \
+      --data-value dockerrepo="${KO_DOCKER_REPO}" \
+      --data-value prowtag="${PROW_TAG}" \
+      --data-value influxurl="${INFLUX_URL}" \
+      --data-value influxtoken="${INFLUX_TOKEN}" \
+      --output-files "${ARTIFACTS}/mako-overlay"
```

Comment:
looks like this is called several times (with a different first file)... maybe a helper function to reduce duplication?

Reply:
`run_ytt` is a helper func in e2e_common, but makes sense to add another wrapper helper like you suggested